sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -20,6 +22,9 @@ namespace Azure.Core.Serialization
     /// This and related types are not intended to be mocked.
     /// </summary>
     [DebuggerDisplay("{DebuggerDisplay,nq}")]
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
```

Comment:
Should we use `NET6_0_OR_GREATER`?

Reply:
The file [`TrimmingAttributes.cs`](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/TrimmingAttribute.cs#L54) added by @m-nash defines this attribute in the `System.Diagnostics.CodeAnalysis` namespace. He might be able to explain it a little better, but this attribute was already defined in NET 5, but just not for classes. My understanding is that keeping it to !NET5_0 allows it to get picked up by netstandard2.0. There was a problem I ran into when using NET6_0_OR_GREATER where OpenTelemetry wasn't targeting net6.0, so the attributes weren't getting picked up. That doesn't happen if using this approach.

Reply:
It's a "look-alike" like nullable attributes; though, I'm surprised: does it not have to be in the same namespace as what .NET defines? IIRC, other look-alikes had to (or maybe they didn't all along?!).

Reply:
@heaths it is like the nullable attributes, the trimmingattribute file defines it in the`System.Diagnostics.CodeAnalysis` namespace which is the same as .NET.